applications/launchpad/gui-react/src/ErrorBoundary.tsx

```diff
@@ -0,0 +1,299 @@
+
+import linksConfig from './config/links'
+import SvgTariLaunchpadLogo from './styles/Icons/TariLaunchpadLogo'
+import SvgTBotBase from './styles/Icons/TBotBase'
+
+interface ErrorBoundaryProps {
+  children?: ReactNode
+}
+
+interface ErrorBoundaryState {
+  error?: string
+  showDetails: boolean
+}
+
+const styles = {
+  windowContainer: {
+    width: '100%',
+    height: '100%',
+    boxSizing: 'border-box',
+    display: 'flex',
+    flexDirection: 'column',
+  },
+  titlebar: {
+    width: '100%',
+    minHeight: 60,
+    background: '#FAFAFA',
+    padding: '16px 40px',
+    boxSizing: 'border-box',
+    display: 'flex',
+    alignItems: 'center',
+  },
+  titleBarButtons: {
+    display: 'flex',
+    alignItems: 'center',
+    marginRight: 32,
+  },
+  windowBtn: {
+    margin: 0,
+    padding: 2,
+    width: 14,
+    height: 14,
+    borderRadius: '50%',
+    boxShadow: 'none',
+    border: '1px solid transparent',
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'center',
+    marginRight: 4,
+    marginLeft: 4,
+    cursor: 'pointer',
+    boxSizing: 'border-box',
+  },
+  mainContainer: {
+    display: 'flex',
+    flex: 1,
+    flexDirection: 'column',
+    alignItems: 'center',
+    justifyContent: 'center',
+    padding: 40,
+    boxSizing: 'border-box',
+  },
+  content: {
+    width: '100%',
+    maxWidth: 700,
+    minHeight: 100,
+    padding: 40,
+    borderRadius: 6,
+    border: '1px solid #EDECEE',
+  },
+  header: {
+    display: 'flex',
+    alignItems: 'center',
+    marginBottom: 32,
+  },
+  heading: {
+    fontFamily: 'AvenirMedium',
+    marginLeft: 16,
+    marginTop: 8,
+    marginBottom: 0,
+  },
+  mainMessageContainer: {
+    paddingTop: 16,
+    paddingBottom: 16,
+  },
+  text: {
+    fontFamily: 'AvenirMedium',
+    color: '#716A78',
+    maxWidth: 600,
+    lineHeight: '160%',
+  },
+  linkInText: {
+    fontFamily: 'AvenirMedium',
+    color: '#716A78',
+  },
+  buttons: {
+    paddingTop: 16,
+    paddingBottom: 16,
+    display: 'flex',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+    columnGap: 10,
+  },
+  button: {
+    display: 'flex',
+    flexDirection: 'row',
+    justifyContent: 'center',
+    alignItems: 'center',
+    padding: '6px 24px',
+    paddingTop: 10,
+    gap: 10,
+    background: 'linear-gradient(239.91deg, #9330FF 0%, #593A9B 131%)',
+    borderRadius: 8,
+    borderWidth: 0,
+    fontFamily: 'AvenirMedium',
+    boxShadow: 'none',
+    fontSize: 14,
+    color: '#fff',
+    cursor: 'pointer',
+  },
+  detailsContainer: {
+    display: 'none',
+    background: '#FAFAFA',
+    borderRadius: 6,
+    border: '1px solid #EDECEE',
+    padding: 12,
+    maxHeight: 200,
+    overflow: 'auto',
+  },
+  detailsText: {
+    fontSize: 14,
+    fontFamily: 'AvenirMedium',
+    opacity: 0.7,
+  },
+}
+
+/**
+ * Catches the app exceptions.
+ *
+ * This implementation tries to NOT depend on anything, ie. styled-components and themes.
```

Comment:
How does work for i18n? Just hard-coded English text for now?

Reply:
Yes, I made this decision to minimise the number of _dependencies_. Now, not using locales is not a real case for us, because it's just plain object, but if any i18n will be added, then I think it's still good to not be dependent on specific library.